docs/Development-Workflow-Guide.md

```diff
@@ -0,0 +1,285 @@
+sudo systemctl enable --now libvirtd
+```
+
+### 1.2 Install kcli
+
+```bash
+ curl -s https://raw.githubusercontent.com/karmab/kcli/main/install.sh | bash
+```
+
+### 1.3 Configure VM profiles
+
+Define a VM profiles. If the images are missing, kcli will fetch them.
+
+```bash
+mkdir -p ${HOME}/.kcli
+touch ${HOME}/.kcli/profiles.yml
+```
+
+edit {HOME}/.kcli/profiles.yml
+
+```yaml
+# ${HOME}/.kcli/profiles.yml
+
+dev-24.04:
+  image: ubuntu2404
+  numcpus: 4
+  memory: 16384
+  disks:
+    - size: 40
+```
+
+### 1.4 Create a VM
+
+Libvirt stores VM disks in storage pools. Create the default pool at /var/lib/libvirt/images and grant your user write access
+
+```bash
+sudo kcli create pool -p /var/lib/libvirt/images default
+sudo setfacl -m u:$(id -un):rwx /var/lib/libvirt/images
+```
+
+Then create a development VM from the profiles above.
+
+```bash
+kcli create vm -p dev-24.04 {your_VMname}
+```
+
+### 1.5 List VMs
+
+Check that the VM is running and reachable.
+
+```bash
+kcli list vm
+# (optional) show more details
+kcli info vm {your_VMname}
+```
+
+### 1.6 Install Docker (for image/rootfs builds)
+
+Docker is required later by osbuilder when building a image.
+Install Docker using the [official documentation](https://docs.docker.com/engine/install/ubuntu/#install-using-the-convenience-script).
+
+---
+
+## 2. Configure the VM environment
+
+### 2.1 Start the VM (if needed) and SSH in
+
+Start the VM if it is stopped, then connect. If it’s already running, kcli will report it and do nothing.
+
+```bash
+# Power on the VM if not running
+kcli start vm {your_VMname}
+
+# (optional) verify state; it should show "running"
+kcli list vm
+
+# Connect via SSH (kcli uses the injected key/user from the cloud image)
+kcli ssh {your_VMname}
+```
+
+Note: It can take a short while after boot for SSH to become available. If SSH fails initially, wait a few seconds and retry.
+
+### 2.2 Set environment variables and aliases
+
+These variables control which container engine and Kubernetes flavor the helper script deploys. The aliases speed up common tasks.
+Edit `~/.bashrc` file and add:
+
+```bash
+export EDITOR="vim" # fidencio's preferred editor
+export CONTAINER_ENGINE="containerd" # whether to use containerd or crio as the container engine -- fidencio's preferred choice is containerd
+export CONTAINER_ENGINE_VERSION="v2.2" # the version of the containerd to be used -- fidencio's preferred choice is always the latest released version
+export KUBERNETES="vanilla" # it can be vanilla (as in, kubeadm deploted), k3s, rke2, k0s, or microk8s -- fidencio's preferred choise is to use the vanilla option
+
+alias install_utils='sudo apt-get update && sudo apt-get -y install vim tig build-essential'
+alias clone_kata='git clone https://github.com/kata-containers/kata-containers $HOME/kata-containers'
+alias deploy_k8s='pushd $HOME/kata-containers && bash tests/integration/kubernetes/gha-run.sh deploy-k8s && popd'
+alias setup_all='clone_kata && deploy_k8s'

```

Comment:
IMHO - given that they are only used once, I don't think these aliases are particularly helpful and there would be more simplicity in just listing the Steps to Clone and then deploy in 2.3